.github/workflows/docs-lint.yaml

```diff
@@ -0,0 +1,25 @@
+name: Check Markdown links
+
+on:
+  push:
```

Comment:
Do we want to run this separately on push to main? We have a [main.yaml](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/blob/main/.github/workflows/main.yml) specifically for running tests after merge to main. Let's follow same suite. wdyt?

We did this to reduce clutter when seeing jobs that run post merge and be only explicit about different workflows during PR commits. 

Example, see this [post merge Action](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/actions/runs/1443995896) vs [detailed runs](https://github.com/vmware-tanzu/cluster-api-provider-bringyourownhost/actions/runs/1444151529)

Reply:
I think it's more appropriate to run only on `pull_request` and not necessary on push to the main branch. The reason is that it should only run this action when there is a `.md` file in the PR and skip running when the `.md` file is not committed in the PR.

This will save us some time while running many other jobs for each PR.